state_machine/applications/flight/lib/radio_utils/commands.py

```diff
@@ -133,15 +134,32 @@ def delete_file(task, file):
     except Exception as e:
         task.debug(f'Error deleting file: {e}')
         _downlink(f'Error deleting file: {e}', priority=9)
-# Helper functions
+
+async def reload(task):
+    """Reloads the flight software
+
+    :param task: The task that called this function
+    """
+    task.debug('Reloading')
+    msg = bytearray([headers.DEFAULT])
+    msg.append(b'reset')
+    await cubesat.radio.send(data=msg)

```

Comment:
Doesn't seem like we'd want the software to switch to a different thread here. Is there a way to not `await` this, but force the software to wait until it completes?

Reply:
Yes, will require a slight rewrite of the driver, will address in separate pull.